File: app/src/main/java/com/ivanovsky/passnotes/presentation/unlock/UnlockContract.java
package com.ivanovsky.passnotes.presentation.unlock;

import android.arch.lifecycle.LiveData;

import com.ivanovsky.passnotes.data.entity.UsedFile;
import com.ivanovsky.passnotes.presentation.core.BasePresenter;
import com.ivanovsky.passnotes.presentation.core.BaseView;
import com.ivanovsky.passnotes.presentation.core.ScreenState;

import java.io.File;
import java.util.List;

class UnlockContract {

	interface View extends BaseView<Presenter> {
	}

	interface Presenter extends BasePresenter {
		void loadData();
		LiveData<List<UsedFile>> getRecentlyUsedFilesData();
		LiveData<ScreenState> getScreenStateData();
		LiveData<Void> getShowGroupsScreenAction();
		LiveData<Void> getShowNewDatabaseScreenAction();
		LiveData<Void> getHideKeyboardAction();
		void onUnlockButtonClicked(String password, File dbFile);
	}
}

File: app/src/main/kotlin/com/ivanovsky/passnotes/presentation/unlock/UnlockContract.kt
package com.ivanovsky.passnotes.presentation.unlock

import com.ivanovsky.passnotes.data.entity.FileDescriptor
import com.ivanovsky.passnotes.presentation.core.BasePresenter
import com.ivanovsky.passnotes.presentation.core.BaseView

class UnlockContract {

    interface View : BaseView<Presenter> {
        fun setRecentlyUsedItems(items: List<UnlockFragment.DropDownItem>)
        fun setSelectedRecentlyUsedItem(position: Int)
        fun showGroupsScreen()
        fun showNewDatabaseScreen()
        fun showOpenFileScreen()
        fun showSettingScreen()
        fun showAboutScreen()
        fun showDebugMenuScreen()
    }

    interface Presenter : BasePresenter {

        fun loadData(resetSelection: Boolean)
        fun onRecentlyUsedItemSelected(position: Int)
        fun onUnlockButtonClicked(password: String)
        fun onOpenFileMenuClicked()
        fun onSettingsMenuClicked()
        fun onAboutMenuClicked()
        fun onDebugMenuClicked()
        fun onFilePicked(file: FileDescriptor)
        fun closeActiveDatabaseIfNeed()
    }
}